### PR TITLE
Fixes for Bittrex and Kraken after testing live orders.

### PR DESCRIPTION
**Pull request by EmergentCybernetics**

### CODE REVIEW

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs [hunk 1]
```diff
@@ -252,7 +252,25 @@ protected override async Task<ExchangeTicker> OnGetTickerAsync(string marketSymb
 		#region OrderBooks
 		protected override async Task<ExchangeOrderBook> OnGetOrderBookAsync(string marketSymbol, int maxCount = 25)
 		{
-			JToken token = await MakeJsonRequestAsync<JToken>("/markets/" + marketSymbol + "/orderbook" + marketSymbol + "&depth=" + maxCount);
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs:255]: This URL is invalid. Fixed below.

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs [hunk 2]
```diff
@@ -252,7 +252,25 @@ protected override async Task<ExchangeTicker> OnGetTickerAsync(string marketSymb
...
+			// Bittrex API allowed values are [1, 25, 500], default is 25.
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs:255]: See: https://bittrex.github.io/api/v3#operation--markets--marketSymbol--orderbook-get

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs [hunk 3]
```diff
@@ -370,7 +388,7 @@ protected override async Task<ExchangeOrderResult> OnPlaceOrderAsync(ExchangeOrd
 			if (order.OrderType == ExchangeSharp.OrderType.Limit)
 			{
 				orderParams.Add("limit", orderPrice);
-				//orderParams.Add("timeInForce", "GOOD_TIL_CANCELLED");
+				orderParams.Add("timeInForce", "GOOD_TIL_CANCELLED");
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI.cs:391]: Bittrex V3 API is saying this is a required parameter now for limit orders.

File: src/ExchangeSharp/API/Exchanges/Kraken/ExchangeKrakenAPI.cs [hunk 4]
```diff
@@ -749,13 +749,17 @@ protected override async Task<IEnumerable<MarketCandle>> OnGetCandlesAsync(strin
...
+
 			object nonce = await GenerateNonceAsync();
 			Dictionary<string, object> payload = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
 			{ { "pair", order.MarketSymbol }, { "type", (order.IsBuy ? "buy" : "sell") }, { "ordertype", order.OrderType.ToString().ToLowerInvariant() }, { "volume", order.RoundAmount().ToStringInvariant() }, { "trading_agreement", "agree" }, { "nonce", nonce }
 			};
 			if (order.OrderType != OrderType.Market)
 			{
-				payload.Add("price", order.Price.ToStringInvariant());
+				int precision = BitConverter.GetBytes(Decimal.GetBits((decimal)market.PriceStepSize)[3])[2];
+				payload.Add("price", Math.Round(order.Price, precision).ToStringInvariant());
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Kraken/ExchangeKrakenAPI.cs:762]: This rounding avoids an error returned by Kraken's API that `price` can only have so many decimals.

**Review by vslee — APPROVED** [review]

**Pull request merged**